### ISSUE TITLE
[plugin-build-web] add support for "browser" field in package.json 

### ISSUE THREAD

**Issue by gr2m** [pikapkg/builders#84]
This is a follow up to https://github.com/pikapkg/builders/pull/46, which added support of the `"browser"` field in dependencies, but it does not support the `"browser"` field in the project's own `package.json`

I got it working by editing `node_modules/@pika/plugin-build-web/dist-node/index.js`

```diff
  const result = await rollup.rollup({
    input: path.join(out, "dist-src/index.js"),
-   plugins: [],
+   plugins: [
+     require("rollup-plugin-node-resolve")({
+       browser: true
+     })
+   ],
    onwarn: (warning, defaultOnWarnHandler) => {
      // // Unresolved external imports are expected
      if (
        warning.code === "UNRESOLVED_IMPORT" &&
        !(warning.source.startsWith("./") || warning.source.startsWith("../"))
      ) {
        return;
      }

      defaultOnWarnHandler(warning);
    }
  });
```

Then in the package.json, the `"browser"` field needs to point to paths in `./pkg/dist-web/*`, e.g. my example is

```json
  "browser": {
    "./pkg/dist-src/get-token.js": "./pkg/dist-src/get-token-browser.js"
  },
```

**Comment by gr2m** [comment]
Would this break usage? I think it might now resolve external dependencies ... but I don't know rollup very well

**Comment by gr2m** [comment]
It should also be mentioned that `"browser": "./pkg/dist-src/browser.js"` will not work, unless this line gets changed to load the path defined in the `"browser"` key instead if it is a string.

Would look something like that?

```
-   input: path.join(out, "dist-src/index.js")
+   input: path.join(out, typeof pkg.browser === 'string' ? pkg.browser : "dist-src/index.js")
```

**Comment by gr2m** [comment]
I also found that in order to make the web-build work with legacy libraries that have the commonJS export, I also need to add the [rollup-plugin-commonjs](https://github.com/rollup/rollup-plugin-commonjs) package and load it as a 2nd plugin

```js
    input: path.join(out, "dist-src/index.js"),
-   plugins: [],
+   plugins: [
+     require("rollup-plugin-node-resolve")({
+       browser: true
+     }),
+     require("rollup-plugin-commonjs")()
+   ],
    onwarn: (warning, defaultOnWarnHandler) => {
      // // Unresolved external imports are expected
      if (
        warning.code === "UNRESOLVED_IMPORT" &&
        !(warning.source.startsWith("./") || warning.source.startsWith("../"))
      ) {
        return;
      }

      defaultOnWarnHandler(warning);
    }
  });
```

**Comment by gr2m** [comment]
@FredKSchott could you give me feedback on this? I'm happy to send a PR, I'm using my own fork with the changes for several packages of mine, it works well

**Comment by FredKSchott** [comment]
Sorry for the lack of response here as we've worked through https://github.com/pikapkg/builders/issues/90. Are you still interested in tackling what was outlined here? https://github.com/pikapkg/builders/issues/90#issuecomment-542450467 If not, I'll implement in the next day or two.

**Comment by gr2m** [comment]
I can wait. I published my test packages with the `"browser:module"` key for testing

- http://pkg.fyi/@gr2m/universal-dependency
- http://pkg.fyi/@gr2m/universal-sub-dependency

**Comment by FredKSchott** [comment]
Resolved via ac46b76fd5775fddefd180ca515fe7dc46d03b4f

**Comment by gr2m** [comment]
Do you use it in your CDN builds yet? I've created a new version `2.0.1` for both my test packages, but the code is from the `"module"` entry point for both, not the `"browser:module"` entry point: 
https://cdn.pika.dev/-/@gr2m/universal-dependency/2.0.1/dist-es2017/universal-dependency.min.js

```js
import 'https://cdn.pika.dev/_/polyfill.js';

function e() {
    console.log('[@gr2m/universal-dependency] "module" entry point')
    console.log('[@gr2m/universal-sub-dependency] "module" entry point')
}
export {
    e as dependency
};
```
